remoteappmanager/docker/container.py

```diff
@@ -32,12 +32,22 @@ class Container(HasTraits):
     #: the id that will go in the URL of the container
     url_id = Unicode()
 
+    #: The absolute base prefix path for the container.
+    base_urlpath = Unicode(None, allow_none=True)
+
     @property
-    def url(self):
+    def urlpath(self):
         """Returns the relative url of the Container."""
         return "containers/{}".format(self.url_id)
 
     @property
+    def absurlpath(self):
```

Comment:
>  I would think a get_container_url as a BaseHandler method maybe good place for it.

You can't. The ReverseProxy class has an add_container() method. you pass the container, so the container must know its full absolute path. Unless we make the ReverseProxy aware of the application, which also sounds weird, or we pass the base path to the add_container method, or, as I said, pass a base path to the ReverseProxy class so that it gets appended to every request (and the container only knows its relative path).

I don' t have a strong opinion on any of these options. They are all weird in some way or another.